browser/ui/views/frame/vertical_tab_strip_region_view.cc

```diff
@@ -140,6 +148,224 @@ class CustomScrollView : public views::ScrollView {
+    const auto text_font = text_->font_list();
+    text_->SetFontList(
+        text_font.DeriveWithSizeDelta(kFontSize - text_font.GetFontSize()));
+
+    shortcut_text_ = AddChildView(std::make_unique<views::Label>());
+    shortcut_text_->SetHorizontalAlignment(
+        gfx::HorizontalAlignment::ALIGN_RIGHT);
+    shortcut_text_->SetVerticalAlignment(gfx::VerticalAlignment::ALIGN_MIDDLE);
+    shortcut_text_->SetVisible(false);
+    const auto shortcut_font = shortcut_text_->font_list();
+    text_->SetFontList(shortcut_font.DeriveWithSizeDelta(
+        kFontSize - shortcut_font.GetFontSize()));
+
+    SetTooltipText(l10n_util::GetStringUTF16(IDS_TOOLTIP_NEW_TAB));
+    SetAccessibleName(l10n_util::GetStringUTF16(IDS_ACCNAME_NEWTAB));
+    SetShortcutText(shortcut_text);
+
+    SetImageVerticalAlignment(
+        views::ImageButton::VerticalAlignment::ALIGN_MIDDLE);
+  }
+
+  ~VerticalTabNewTabButton() override = default;
+
+  // BraveNewTabButton:
+  SkPath GetBorderPath(const gfx::Point& origin,
+                       float scale,
+                       bool extend_to_top) const override {
+    auto contents_bounds = GetContentsBounds();
+    const float radius = tabs::kUnpinnedTabBorderRadius * scale;
+    SkPath path;
+    const gfx::Rect path_rect(origin.x(), origin.y(),
+                              contents_bounds.width() * scale,
+                              contents_bounds.height() * scale);
+    path.addRoundRect(RectToSkRect(path_rect), radius, radius);
+    path.close();
+    return path;
+  }
+
+  void PaintIcon(gfx::Canvas* canvas) override {
+    // Revert back the offset set by NewTabButton::PaintButtonContents(), which
+    // is the caller of this method.
+    gfx::ScopedCanvas scoped_canvas(canvas);
+    canvas->Translate(-GetContentsBounds().OffsetFromOrigin());
+
+    // Set additional horizontal inset for '+' icon.
+    if (region_view_->state() ==
+        VerticalTabStripRegionView::State::kCollapsed) {
+      // // When |text_| is empty, the vertical tab strip could be minimized.
+      // // In this case we should align icon center.
+      // // TODO(sko) Should we keep this padding when it's transitioned to
+      // // floating mode?
+      SetImageHorizontalAlignment(
+          views::ImageButton::HorizontalAlignment::ALIGN_CENTER);
+    } else {
+      SetImageHorizontalAlignment(
+          views::ImageButton::HorizontalAlignment::ALIGN_LEFT);
+      canvas->Translate({6, 0});
+    }
+
+    // Use vector icon
+    ImageButton::PaintButtonContents(canvas);
+  }
+
+  gfx::Insets GetInsets() const override {
+    return gfx::Insets(tabs::kMarginForVerticalTabContainers);
+  }
+
+  void OnPaintFill(gfx::Canvas* canvas) const override {
+    if (tab_strip()
+            ->GetCustomBackgroundId(BrowserFrameActiveState::kUseCurrent)
+            .has_value()) {
+      BraveNewTabButton::OnPaintFill(canvas);
+      return;
+    }
+
+    auto* cp = GetColorProvider();
+    DCHECK(cp);
```

Comment:
I saw a thread that Chromium's style guide has changed to prefer `CHECK` but I don't know what Brave's stance is 
https://bravesoftware.slack.com/archives/C7VLGSR55/p1679324978507629

Reply:
Oh, missed that. Thanks for letting me know that!